### DISCUSSION TITLE
JCB v3.1.19

### DISCUSSION THREAD

**Discussion by Llewellynvdm** [joomengine/Joomla-Component-Builder#950] [Announcements]
Another Stable JCB release [JCB v3.1.19](https://git.vdm.dev/joomla/pkg-component-builder/archive/master.zip)

We have fixed a number of bugs, and believe the is the most stable version since the refactoring started.

- We fixed [#972](https://git.vdm.dev/joomla/Component-Builder/issues/972) so that custom code (in the header) will be added after the power namespaces.
- We added a message to show when a server move failed.
- We fixed the BaseConfig to not use '_' as separator.
- We fixed the footable loading issue.
- We removed the need for passing placeholders by reference.
- We added the option to generate a CHANGELOG.
- We fixed the server class to load new client if server details changed.
- We fixed the readme placeholder issue [#978](https://git.vdm.dev/joomla/Component-Builder/issues/978).
- We fixed the empty server url issue [#978](https://git.vdm.dev/joomla/Component-Builder/issues/978).
- Fixed Package import to now use the phplibsec version 3 [#979](https://git.vdm.dev/joomla/Component-Builder/issues/979).

ENJOY! We are moving forward!

**Comment by andiamine** [comment]
Thanks, great job
